src/write.c

```diff
@@ -886,6 +886,56 @@ static int avifQualityToQuantizer(int quality, int minQuantizer, int maxQuantize
     return quantizer;
 }
 
+// Adds the items for a single cell or a grid of cells.
+// Outputs the mainItemID, being the only cell if there is exactly one, or the grid item for multiple cells.
+static avifResult avifEncoderAddImageItems(avifEncoder * encoder,
+                                           uint32_t gridCols,
+                                           uint32_t gridRows,
+                                           uint32_t gridWidth,
+                                           uint32_t gridHeight,
+                                           avifBool alpha,
+                                           uint16_t * mainItemID)
+{
+    const uint32_t cellCount = gridCols * gridRows;
+    const char * infeName = alpha ? "Alpha" : "Color";
+    const size_t infeNameSize = 6;
+
+    if (cellCount > 1) {
+        avifEncoderItem * gridItem = avifEncoderDataCreateItem(encoder->data, "grid", infeName, infeNameSize, 0);
+        avifWriteGridPayload(&gridItem->metadataPayload, gridCols, gridRows, gridWidth, gridHeight);
+        gridItem->alpha = alpha;
+        gridItem->gridCols = gridCols;
+        gridItem->gridRows = gridRows;
+        gridItem->gridWidth = gridWidth;
+        gridItem->gridHeight = gridHeight;
+        *mainItemID = gridItem->id;
+    }
+
+    for (uint32_t cellIndex = 0; cellIndex < cellCount; ++cellIndex) {
+        avifEncoderItem * item = avifEncoderDataCreateItem(encoder->data, "av01", infeName, infeNameSize, cellIndex);
+        item->codec = avifCodecCreate(encoder->codecChoice, AVIF_CODEC_FLAG_CAN_ENCODE);
+        if (!item->codec) {
+            // Just bail out early, we're not surviving this function without an encoder compiled in
+            return AVIF_RESULT_NO_CODEC_AVAILABLE;
+        }
+        item->codec->csOptions = encoder->csOptions;
+        item->codec->diag = &encoder->diag;
+        item->alpha = alpha;
+        item->extraLayerCount = encoder->extraLayerCount;
+
+        if (cellCount > 1) {
+            item->dimgFromID = *mainItemID;
+            item->hiddenImage = AVIF_TRUE;
+        } else {
+            *mainItemID = item->id;
+        }
+    }
+
+    // The mainItemID output argument has the type uint16_t* instead of avifEncoderItem*
+    // because the pointer may be invalidated by a call to avifEncoderDataCreateItem().
```

Comment:
"the pointer" is ambiguous because both `uint16_t*` and `avifEncoderItem*` are pointers.

I guess we can change "the pointer" to "the avifEncoderItem pointer".